src/components/StyleGuide/Colors.stories.mdx

```diff
@@ -9,372 +9,669 @@ import DSProvider from "../../theme/provider";
+      dataTextColor: colorContrastData.dark.ui.gray.xDark.dataTextColor,
+      textColor: colorContrastData.ui.gray.xDark.textColor,
+    },
+    {
+      colorSource: "ui.gray.x-dark",
+      name: "dark.ui.border.default",
+      dataBlackColor: colorContrastData.ui.gray.xDark.dataBlackColor,
+      dataWhiteColor: colorContrastData.ui.gray.xDark.dataWhiteColor,
+      dataHeadingColor: colorContrastData.dark.ui.gray.xDark.dataHeadingColor,
+      dataTextColor: colorContrastData.dark.ui.gray.xDark.dataTextColor,
+      textColor: colorContrastData.ui.gray.xDark.textColor,
+    },
+    {
+      colorSource: "ui.gray.medium",
+      name: "dark.ui.border.hover",
+      dataBlackColor: colorContrastData.ui.gray.medium.dataBlackColor,
+      dataWhiteColor: colorContrastData.ui.gray.medium.dataWhiteColor,
+      dataHeadingColor: ["1.55", "Fail", "Fail"],
+      dataTextColor: ["1.00", "Fail", "Fail"],
+      textColor: colorContrastData.ui.gray.medium.textColor,
+    },
+    {
+      colorSource: "ui.gray.x-dark",
+      name: "dark.ui.disabled.primary",
+      dataBlackColor: colorContrastData.ui.gray.xDark.dataBlackColor,
+      dataWhiteColor: colorContrastData.ui.gray.xDark.dataWhiteColor,
+      dataHeadingColor: colorContrastData.dark.ui.gray.xDark.dataHeadingColor,
+      dataTextColor: colorContrastData.dark.ui.gray.xDark.dataTextColor,
+      textColor: colorContrastData.ui.gray.xDark.textColor,
+    },
+    {
+      colorSource: "ui.gray.xx-dark",
+      name: "dark.ui.disabled.secondary",
+      dataBlackColor: colorContrastData.ui.gray.xxDark.dataBlackColor,
+      dataWhiteColor: colorContrastData.ui.gray.xxDark.dataWhiteColor,
+      dataHeadingColor: colorContrastData.dark.ui.gray.xxDark.dataHeadingColor,
+      dataTextColor: colorContrastData.dark.ui.gray.xxDark.dataTextColor,
+      textColor: colorContrastData.ui.gray.xxDark.textColor,
+    },
+    {
+      name: "dark.ui.error.primary",
+      dataBgPageColor: ["4.60", "AA", "AAA"],
+      dataBgDefaultColor: ["4.32", "Fail", "AA"],
+      dataBlackColor: ["5.16", "AA", "AAA"],
+      dataWhiteColor: ["4.01", "Fail", "AA"],
+      textColor: "ui.black",
+    },
+    {
+      name: "dark.ui.error.secondary",
+      dataBgPageColor: ["7.25", "AAA", "AAA"],
+      dataBgDefaultColor: ["6.80", "AA", "AAA"],
+      dataBlackColor: ["8.12", "AAA", "AAA"],
+      dataWhiteColor: ["2.58", "Fail", "Fail"],
+      textColor: "ui.black",
+    },
+    {
+      name: "dark.ui.focus",
+      dataBgPageColor: ["4.85", "AA", "AAA"],
+      dataBgDefaultColor: ["4.56", "AA", "AAA"],
+      dataBlackColor: ["5.44", "AA", "AAA"],
+      dataWhiteColor: ["3.86", "Fail", "AA"],
+      textColor: "ui.black",
+    },
+    {
+      name: "dark.ui.link.primary",
+      dataBgPageColor: ["4.82", "AA", "AAA"],
+      dataBgDefaultColor: ["4.53", "AA", "AAA"],
+      dataBlackColor: ["5.41", "AA", "AAA"],
+      dataWhiteColor: ["2.82", "Fail", "AA"],
```

Comment:
I had this wrong in the Figma.  The contrast ratio for `dark.ui.link.primary` should be `3.88`.